src/azure.rs

```diff
@@ -447,11 +460,13 @@ fn convert_object_meta(blob: Blob) -> Result<ObjectMeta> {
         .try_into()
         .expect("unsupported size on this platform");
 
-    Ok(ObjectMeta {
+    // This is needed to filter out gen2 directories
+    // https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-known-issues#blob-storage-apis
+    Ok((size > 0).then(|| ObjectMeta {
```

Comment:
Another drive-by-fix, this logic is moved to convert_object_meta so that it is applied consistently to all requests